crates/emath/src/ts_transform.rs

```diff
@@ -0,0 +1,91 @@
+use crate::{Pos2, Rect, Vec2};
+
+/// Linearly transforms positions via a translation, then a scaling.
+///
+/// [`TSTransform`] first translates points, then scales them with the scaling origin
+/// at `0, 0` (the top left corner)
+#[repr(C)]
+#[derive(Clone, Copy, Debug, PartialEq)]
+#[cfg_attr(feature = "serde", derive(serde::Deserialize, serde::Serialize))]
+#[cfg_attr(feature = "bytemuck", derive(bytemuck::Pod, bytemuck::Zeroable))]
+pub struct TSTransform {
+    /// Translation amount.
+    pub translation: Vec2,
+
+    /// Scaling amount after translation, scaled around (0, 0).
+    pub scaling: f32,
+}
```

Comment:
We should switch this. The common convention is `T * S * v`; see for instance https://docs.rs/glam/latest/glam/f32/struct.Affine2.html#method.from_scale_angle_translation

```suggestion
pub struct TSTransform {
    /// Scaling applied first, scaled around (0, 0).
    pub scaling: f32,
    
    /// Translation amount, applied after translation.
    pub translation: Vec2,
}
```